src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -26,38 +29,72 @@ public OrasClient(string imageReference)
             _imageReference = imageReference;
         }
 
-        public async Task PullImageAsync(string outputFolder)
+        public async Task<ManifestWrapper> PullImageAsync(string outputFolder)
         {
+            if (Directory.Exists(outputFolder) && Directory.GetFileSystemEntries(outputFolder).Length != 0)
+            {
+                throw new OCIClientException(TemplateManagementErrorCode.OrasProcessFailed, "The output folder is not empty.");
+            }
+
             string command = $"pull  \"{_imageReference}\" -o \"{outputFolder}\"";
-            await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
+            string output = await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
+            string digest;
+            try
+            {
+                digest = _digestRegex.Matches(output)[0].Groups["hex"].ToString();
+            }
+            catch (Exception ex)
+            {
+                throw new OCIClientException(TemplateManagementErrorCode.OrasProcessFailed, "Return image digest failed.", ex);
+            }
+
+            // Oras will create output folder if not existed.
+            // Therefore, the output folder should exist if pull succeed.
+            if (!Directory.Exists(outputFolder))
+            {
+                throw new OCIClientException(TemplateManagementErrorCode.OrasProcessFailed, "Pull image failed or image is empty.");
+            }
+
+            try
+            {
+                string manifestContent = LoadManifestContentFromCache(digest);
+                var manifest = JsonConvert.DeserializeObject<ManifestWrapper>(manifestContent);
+                File.WriteAllText(Path.Combine(outputFolder, Constants.ManifestFileName), LoadManifestContentFromCache(digest));
```

Comment:
do we need to write the content to this file?

Reply:
After offline discussion, changed it to write the file by caller.